addons/finalAlert/finalAlert.lua

```diff
@@ -0,0 +1,303 @@
+Copyright © 2022, Godchain
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of <addon name> nor the
+      names of its contributors may be used to endorse or promote products
+      derived from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL Godchain BE LIABLE FOR ANY
+DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+_addon.name = "finalAlert"
+_addon.author = "Godchain (Asura)"
+_addon.version = "1.3"
+_addon.commands = {"finalAlert", "fa"}
+
+config = require("config")
+texts = require("texts")
+res = require("resources")
+
+caption = texts.new({})
+
+background_ability = "background_ability"
+background_magic = "background_magic"
+background_interrupt = "background_interrupt"
+background_emphasize = "background_emphasize"
+
+-- Timing
+showing = false
+last_trigger = 0
+
+-- IDs
+weapon_skill_category = 7
+magic_category = 8
+interrupt_id = 28787
+
+defaults = {}
+defaults.x_position = windower.get_windower_settings().x_res / 2
+defaults.y_position = 100
+defaults.background_size = "regular"
+defaults.emphasize = S {}
+defaults.trigger_duration = 3
+defaults.sounds = "on"
```

Comment:
I would use boolean values instead of `on`/`off`, and even rather than binary enum values, like for the `background_size` above. I'd reword it to `large` with `true`/`false` as possible values or `small`/`compact`, with the same (but reversed) values. Users are notoriously clumsy with handling a single set of allowed values, you'll not only get users who misspell the allowed values, but also users who will argue about case insensitivity. The boolean approach saves you from a bunch of possible issues.